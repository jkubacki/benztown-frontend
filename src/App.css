.navbar {
  padding: 0;
  margin-bottom: 1em;
  margin-top: 1em;
}
