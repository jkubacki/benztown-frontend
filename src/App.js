import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';

import Homepage from 'views/Homepage'
import AcceptInvitationPage from 'views/AcceptInvitationPage'
import { Route } from 'components/navigation';

import {
  markAsLoggedIn as markAsLoggedInAction,
  markAsNotLoggedIn as markAsNotLoggedInAction
} from 'actions/auth';
import { getStorageItem} from 'utils';
import { getRootPath, getAcceptInvitationPath } from './constants/paths';

class App extends Component {
  componentWillMount() {
    this.updateClientStatus();
  }

  updateClientStatus() {
    const { markAsLoggedIn, markAsNotLoggedIn } = this.props;
    const token = getStorageItem('token');
    if (token) {
      markAsLoggedIn();
    } else {
      markAsNotLoggedIn();
    }
  }

  render() {
    return (
      <div>
        <Route
          path={getRootPath()}
          exact
          component={Homepage}
        />
        <Route
          path={getAcceptInvitationPath()}
          exact
          component={AcceptInvitationPage}
        />
      </div>
    );
  }
}

export default compose(
  connect(
    null,
    {
      markAsLoggedIn: markAsLoggedInAction,
      markAsNotLoggedIn: markAsNotLoggedInAction
    },
  ),
  withRouter,
)(App);
