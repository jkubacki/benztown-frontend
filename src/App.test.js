import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from 'App';

const props = {
  markAsNotLoggedIn: jest.fn(),
  history: { push: jest.fn(), location: { search: '' } },
  match: {},
};

it('renders without crashing', () => {
  const component = shallow(<App {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
