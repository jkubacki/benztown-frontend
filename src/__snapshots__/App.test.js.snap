// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
>
  <Connect(AppHeader) />
  <Switch>
    <Connect(withRouter(Route))
      component={[Function]}
      exact={true}
      path="/"
    />
    <Connect(withRouter(Route))
      component={[Function]}
      exact={true}
      path="/accept-invitation"
    />
  </Switch>
</div>
`;
